internal/server/handlers.go

```diff
@@ -479,6 +481,8 @@ func (a *API) SignupEnabled(c *gin.Context, _ *api.EmptyRequest) (*api.SignupEna
 }
 
 func (a *API) Signup(c *gin.Context, r *api.SignupRequest) (*api.Identity, error) {
+	c.Set("enableSignup", a.server.options.EnableSignup)
```

Comment:
Could we quick-return false here instead of setting the context?

Reply:
That's fine too. I decided to take the current approach as it consolidates the checking into a single place

Reply:
Not having to use context for this would be nice.  

Not blocking, but I think if the call to `access.SignupEnabled` happened here (instead of from `access.Signup`), we could either pass in the bool as an argument, or do the early return.